commands/system/help.js

```diff
@@ -9,74 +11,170 @@ class Help extends Command {
+        if (c.help.category.toLowerCase() === type.toLowerCase()) {
+          n++;
+        }
+      });
+
+      let num = 0;
+      let output = "";
+      const pg = Number(page) && Number(page) <= Math.ceil(n / perpage) ? Number(page) : 1;
+      for (const c of sorted.values()) {
+        if (c.help.category.toLowerCase() === type.toLowerCase()) {
+          if (c.help.cat === "NSFW" && !message.channel.nsfw) return;
+          if (num < perpage * pg && num > perpage * pg - (perpage + 1)) {
+            if (level < this2.client.levelCache[c.conf.permLevel]) return;
+            output = `${output}\n\`${message.settings.prefix + c.help.name}\` | ${c.help.description.length > 80 ? `${c.help.description.slice(0, 80)}...` : c.help.description}`;
+          }
+          num++;
+        }
+      }
+      r.users.remove(message.author);
+      if (hm === "forward") {
+        if (page > Math.ceil(num / perpage)) return;
+      }
+      if (hm === "backward") {
+        if (page === 0) return;
+      }
+      const helpembed = new MessageEmbed()
+        .setTitle(`Page ${page}/${Math.ceil(num / perpage)} for ${type.toProperCase()}`)
+        .setDescription(`A list of commands in the ${type} category.\n(Total of ${num} commands in this category)\n\nTo get help on a specific command do \`${message.settings.prefix}help <command>\``)
+        .addField("Commands", output)
+        .setColor(message.guild.me.roles.highest.color || 5198940)
+        .setFooter(`Requested by ${message.author.tag}`, message.author.avatarURL());
+      await msg2.edit(helpembed);
+    };
   }
 
-  async run(message, [type, page], level) {
+  async run(message, [type], level) {
+    let page = 1;
+    let n = 0;
+    let num = 0;
 
     const embed = new MessageEmbed()
       .setTimestamp()
       .setColor(message.guild.me.roles.highest.color || 5198940)
-      .setFooter(`Requested by ${message.author.tag}`, message.author.avatarURL()); 
+      .setFooter(`Requested by ${message.author.tag}`, message.author.avatarURL());
 
-    let currentCategory = "";
-    const sorted = this.client.commands.sort((p, c) => p.help.category > c.help.category ? 1 :  p.help.name > c.help.name && p.help.category === c.help.category ? 1 : -1 );
+    const currentCategory = "";
+    const last = [];
+    const sorted = this.client.commands.sort((p, c) => (p.help.cat > c.help.category ? 1 : p.help.name > c.help.name && p.help.cat === c.help.category ? 1 : -1));
     if (!type) {
       const description = `Command category list\n\nUse \`${message.settings.prefix}help <category>\` to find commands for a specific category`;
-      const output = sorted.filter(c => !(level < 10 && c.help.category == "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map(c => {
-        const cat = c.help.category.toProperCase();
+      const output = sorted.filter(c => !(level < 10 && c.help.category === "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map((c) => {
```

Comment:
I'd suggest:

Filter the commands with a for-of by the category name + this check. Once it's fully filtered, sort it and send it to the reaction handler.

At this moment, you're sorting, filtering (removing some of the sorted elements), filtering again (removing even more sorted elements). This becomes highly unefficient if a user types a category no command belongs to, and you'd still sort, filter, re-filter, and finally show nothing.